internal/measurexlite/resolver_test.go

```diff
@@ -0,0 +1,163 @@
+package measurexlite
+
+import (
+	"context"
+	"errors"
+	"net"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/ooni/probe-cli/v3/internal/model"
+	"github.com/ooni/probe-cli/v3/internal/model/mocks"
+)
+
+func TestNewTrustedRecursiveResolver(t *testing.T) {
+	var called bool
+	newResolver := func(model.Logger, string) model.Resolver {
+		return &mocks.Resolver{
+			MockLookupHost: func(ctx context.Context, domain string) ([]string, error) {
+				return []string{"1.1.1.1"}, nil
+			},
+			MockLookupHTTPS: func(ctx context.Context, domain string) (*model.HTTPSSvc, error) {
+				return &model.HTTPSSvc{
+					IPv4: []string{"1.1.1.1"},
+				}, nil
+			},
+			MockLookupNS: func(ctx context.Context, domain string) ([]*net.NS, error) {
+				return []*net.NS{
+					{
+						Host: "1.1.1.1",
+					},
+				}, nil
+			},
+			MockCloseIdleConnections: func() {
+				called = true
+			},
+		}
+	}
+
+	t.Run("default URL", func(t *testing.T) {
+		expected := "https://mozilla.cloudflare-dns.com/dns-query"
+		resolver := NewTrustedRecursiveResolver2(&mocks.Logger{}, "")
+		resolvert := resolver.(*TrustedRecursiveResolver2)
+		if resolvert.Url() != expected {
+			t.Fatal("unexpected default url")
+		}
+	})
+
+	t.Run("custom URL", func(t *testing.T) {
+		expected := "https://dns.google/dns-query"
+		resolver := NewTrustedRecursiveResolver2(&mocks.Logger{}, "https://dns.google/dns-query")
+		resolvert := resolver.(*TrustedRecursiveResolver2)
+		if resolvert.Url() != expected {
+			t.Fatal("unexpected default url")
+		}
+	})
+
+	t.Run("with DoH resolver", func(t *testing.T) {
+		resolver := NewTrustedRecursiveResolver2(&mocks.Logger{}, "")
+		resolvert := resolver.(*TrustedRecursiveResolver2)
```

Comment:
```suggestion
		resolver := NewTrustedRecursiveResolver2(&mocks.Logger{}, "").(*TrustedRecursiveResolver2)
```

This should save you a variable.

Reply:
You are right. Thanks!